roboteam_ai/src/coach/heuristics/CoachHeuristics.cpp

```diff
@@ -26,7 +26,8 @@ double CoachHeuristics::calculateCloseToGoalScore(const Vector2 &position) {
 
 /// Gives a higher score if the line between the position and the goal is free
 double CoachHeuristics::calculateShotAtGoalScore(const Vector2 &position, const WorldData &world) {
-    double viewAtGoal = world::field->getPercentageOfGoalVisibleFromPoint(false, position, world)/100;
+    WorldData copy = WorldData({}, world.them, world.ball, world.time);
```

Comment:
This is probably quite expensive since you're creating a worldData object everytime this function is called. Might be better to put the creation of the world  without our robots one abstraction level higher. Especially since most other score functions also ONLY look at their robots.